turbo/stages/headerdownload/header_algos.go

```diff
@@ -587,6 +600,18 @@ func (hd *HeaderDownload) HasLink(linkHash common.Hash) bool {
 	return false
 }
 
+// SaveExternalAnnounce - does mark hash as seen in external announcement
+// only such hashes will broadcast further after
+func (hd *HeaderDownload) SaveExternalAnnounce(hash common.Hash) {
+	hd.lock.Lock()
+	defer hd.lock.Unlock()
+	const maxSawAnnouncesSetSize = 16 * 1024
+	if hd.seenAnnounces.Cardinality() >= maxSawAnnouncesSetSize {
+		hd.seenAnnounces.Pop()
```

Comment:
I think we should something more deterministic than this, otherwise it will be harder to test. Why not use LRU cache, like prefetchedBlocks?

Reply:
make sense. will do

Reply:
changed to lru